os/net/blemgr/bledev.c

```diff
@@ -60,13 +60,14 @@ int trble_scan_data_enque(trble_scanned_device *info)
 
 	int i;
 	uint32_t *u1 = (uint32_t *)info;
-	uint32_t *u2 = (uint32_t *)&g_scan_queue->queue[g_scan_queue->write_index];
-	for (i = 0; i < sizeof(trble_scanned_device) / sizeof(uint32_t); i++) {
+	uint32_t *u2 = (uint32_t *)(g_scan_queue->queue + (g_scan_queue->data_size * g_scan_queue->write_index));
+	for (i = 0; i < g_scan_queue->data_size / sizeof(uint32_t); i++) {
```

Comment:
I think value of data_size can be changed in multi-thread environment, is it intended?
To make loop more stable you had better use below style to prevent error by changed condition. It is very basic style.
```
int a = a.size;  // int a = funcA();
for (i = 0; i < a; .....)
```

Reply:
'data_size' of queue is not changed after initialized first time.